internal/item/index.go

```diff
@@ -0,0 +1,97 @@
+package item
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/ardanlabs/kit/db"
+	"github.com/ardanlabs/kit/db/mongo"
+	"github.com/ardanlabs/kit/log"
+
+	"gopkg.in/mgo.v2"
+)
+
+// =============================================================================
+
+// EnsureIndexes perform index create commands against Mongo for the indexes
+// required by each item type
+func EnsureTypeIndexes(context interface{}, db *db.DB, types map[string]Type) error {
+	log.Dev(context, "Item.EnsureTypeIndexes", "Started")
+
+	var errStr string
+	var indexes []mgo.Index
+
+	// create a list of indexes to create
+	//  note, this list may be incomplete or too aggressive all
+	//  indexes should reference the funcs / cases when they are needed
+
+	// The type + _id index (type limits first)
+	//  Used to find one item by type
+	//  ? - does introducing type make this more efficient than the
```

Comment:
To this question: we can try it both ways.  However, the type_id may be redundant if we end up using the graph for smart indexing, rather than embedding the indices.